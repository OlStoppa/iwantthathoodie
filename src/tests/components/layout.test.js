import React from 'react'
import { StaticQuery } from 'gatsby'
import {render} from '@testing-library/react'
import Context from '../../context/cartContext'
import Layout from '../../components/layout'


  beforeEach(() => {
      StaticQuery.mockImplementationOnce(({ render }) =>
        render({
            "site": {
              "siteMetadata": {
                "title": "this-hoodie"
              }
            }
          }))
  })

  test('layout renders correctly with children',() => {
      const {container, getByText} = render(
        <Context.Provider value={[[]]}>
            <Layout>
                <div>hoodie test</div>
            </Layout>
        </Context.Provider>)
      expect(container).toMatchSnapshot()
      expect(getByText('hoodie test')).toBeInTheDocument()
      
  })