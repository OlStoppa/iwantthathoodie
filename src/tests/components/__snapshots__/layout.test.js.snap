// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout renders correctly with children 1`] = `
<div>
  <div
    class="site-container"
  >
    <header>
      <div
        style="margin: 0px auto; max-width: 1200px; padding: 0.5rem 1.45rem 0rem; display: flex; justify-content: space-between; align-items: center;"
      >
        <div
          class="menu-btn"
        >
          <div
            class="btn-line"
          />
          <div
            class="btn-line"
          />
          <div
            class="btn-line"
          />
        </div>
        <a
          href="/"
          style="color: white; text-decoration: none;"
        >
          <test-file-stub
            classname="logo"
          />
        </a>
        <a
          href="/cart"
        >
          <div
            class="cart-icon__container"
          >
            <span
              class="material-icons"
            >
              shopping_cart
            </span>
          </div>
        </a>
      </div>
      <nav
        class="navbar"
      >
        <ul>
          <a
            href="/collection/pullover"
          >
            <li>
              PULLOVERS
            </li>
          </a>
          <li>
            ZIP-UPS
          </li>
          <a
            href="/collection/t-shirt"
          >
            <li>
              T-SHIRTS
            </li>
          </a>
          <li>
            NEW
          </li>
        </ul>
      </nav>
    </header>
    <div
      class="drawer"
    >
      <div
        class="mobile-nav-container"
      >
        <ul>
          <a
            href="/contact"
          >
            <li>
              Contact Us
            </li>
          </a>
        </ul>
      </div>
    </div>
    <div
      style="margin: 0px auto; max-width: 1200px; padding: 0px 0px 1.45rem 0px; flex-grow: 1;"
    >
      <main>
        <div>
          hoodie test
        </div>
      </main>
    </div>
    <div
      class="footer"
    >
      <div
        class="footer--inner"
      >
        Â© 2019 iwantthathoodie.com
      </div>
    </div>
  </div>
</div>
`;
